py_gtfs_rt_ingestion/dev_test_setup.py

```diff
@@ -0,0 +1,298 @@
+        DEV_ARCHIVE_BUCKET,
+        DEV_ERROR_BUCKET,
+    )
+
+    for bucket in bucket_list:
+        action = None
+        uri_root = f"s3://{os.path.join(bucket, LAMP_PREFIX)}"
+        files_to_delete = get_del_obj_list(bucket, uri_root)
+        if len(files_to_delete) == 0:
+            print("No objects found... skipping bucket.")
+        else:
+            while action not in ("n", "no"):
+                # Print list of bucket objects
+                if action in ("list", "ls"):
+                    print(f"/{'*' * 50}/")
+                    for uri in files_to_delete:
+                        print(f"{uri.replace(uri_root, '')}")
+                    print(f"/{'*' * 50}/")
+                # Proceed with object deletion
+                elif action in ("y", "yes") or args.force_delete:
+                    delete_count = make_del_jobs(files_to_delete, bucket)
+                    # If not all objects deleted, retry
+                    if delete_count < len(files_to_delete):
+                        print(
+                            f"Only {delete_count:,} of {len(files_to_delete):,}"
+                            " deleted... will retry."
+                        )
+                        files_to_delete = get_del_obj_list(bucket, uri_root)
+                    else:
+                        print(f"All {len(files_to_delete)} deleted")
+                        break
+
+                print(
+                    f"{len(files_to_delete):,} objects found, delete? "
+                    "\n"
+                    "yes(y) / no(n) / list(ls)"
+                )
+                action = input()
+
+
+def copy_obj(prefix: str, num_to_copy: int) -> int:
+    """
+    Copy random objects from prefix to dev Ingest bucket
+    """
+    count_objs_to_pull = max(num_to_copy * 10, 1_000)
+    uri_copy_set = set()
+    src_uri_root = f"s3://{os.path.join(GTFS_BUCKET, prefix)}"
+    skip_uri = "https_mbta_busloc_s3.s3.amazonaws.com_prod_TripUpdates_enhanced"
+    print(
+        f"Pulling list of {count_objs_to_pull:,} objects from "
+        f"{src_uri_root} for random sample..."
+    )
+    try:
+        for (uri, size) in file_list_from_s3(GTFS_BUCKET, prefix):
+            # Skip busloc TripUpdates because of S3 permission issues
+            if size > 0 and skip_uri not in uri:
+                uri_copy_set.add(uri)
+            if len(uri_copy_set) == count_objs_to_pull:
+                break
+    except Exception as e:
+        logging.error("Unable to pull file list from %s", src_uri_root)
+        logging.exception(e)
+        return 0
+
+    s3_client = boto3.client("s3")
+    success_count = 0
+    print(
+        f"Starting copy of {num_to_copy} random objects from {src_uri_root} ..."
+    )
+    for uri in random.sample(tuple(uri_copy_set), num_to_copy):
+        key = str(uri).replace(f"s3://{GTFS_BUCKET}/", "")
+        copy_source = {
+            "Bucket": GTFS_BUCKET,
+            "Key": key,
+        }
```

Comment:
Copy random sample of objects pulled. `uri_copy_set` will be a minimum of 1,000 objects and up to 10x the number to be copied.